File: py/10001st_prime.py
#!/usr/bin/env python

import math

num_of_primes = 1  # skipping 2
prime_target = 6
prime_target = 10001

prime_check = 1

def is_prime(num):
  if num % 2 == 0:
    return False
  #print("Checking %i" % num)
  max = int(math.ceil(math.sqrt(num)))
  #print("max = %i" % max)
  for n in range(2, max+1):
    #print("Checking if %i divides %i" % (n, num))
    if num % n == 0:
      return False
  return True

while num_of_primes < prime_target:
  prime_check += 2
  if is_prime(prime_check):
    num_of_primes +=1
    print prime_check

# vim: expandtab ts=2 sw=2 sts=2

File: ruby/10001st_prime.rb
#!/usr/bin/env ruby


# Problem 7 from projecteuler.net
#
# By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see
# that the 6th prime is 13.
#
# What is the 10 001st prime number?
# 

def is_prime? num
  2.upto(Math.sqrt(num)) do |n|
    return false if (num % n == 0)
  end
  return true
end

ctr=1
prime_ctr=0
until prime_ctr == 10001
  ctr += 1
  prime_ctr += 1 if is_prime? ctr
end
  
puts ctr
# vim: ai ts=2 sts=2 et sw=2 ft=ruby
